pkg/knit/cmd/podres.go

```diff
@@ -49,25 +55,56 @@ func NewPodResourcesCommand(knitOpts *KnitOptions) *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			return showPodResources(cmd, opts, args)
 		},
-		Args: cobra.NoArgs,
+		Args: cobra.MaximumNArgs(1),
 	}
 	podRes.Flags().StringVarP(&opts.socketPath, "socket-path", "R", defaultSocketPath, "podresources API socket path.")
 	return podRes
 }
 
+func selectAction(apiName string) (func(cli kubeletpodresourcesv1.PodResourcesListerClient) error, error) {
+	switch apiName {
+	case apiCallList:
+		return func(cli kubeletpodresourcesv1.PodResourcesListerClient) error {
+			resp, err := cli.List(context.TODO(), &kubeletpodresourcesv1.ListPodResourcesRequest{})
+			if err != nil {
+				return err
+			}
+			json.NewEncoder(os.Stdout).Encode(resp)
```

Comment:
can we error out here ? what will happen if we do ?

Reply:
good point, fixing